packages/backend-app-api/src/wiring/createInitializationLogger.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RootLoggerService } from '@backstage/backend-plugin-api';
+
+const LOGGER_INTERVAL_MAX = 60_000;
+
+export function createInitializationLogger(
+  pluginIds: string[],
+  rootLogger?: RootLoggerService,
+): {
+  onPluginStarted(pluginId: string): void;
+  onAllStarted(): void;
+} {
+  const logger = rootLogger?.child({ type: 'initialization' });
+  const starting = new Set(pluginIds);
+  const started = new Set<string>();
+
+  logger?.info(`Plugin initialization started: '${pluginIds.join("', '")}'`);
```

Comment:
maybe break out a `join(iterable)` function and make it instead ```.map(i => `'${i}'`).join(', ')``` just for the unlikely event that there are zero entries ;)